rosidl_cli/rosidl_cli/entry_points.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+try:
+    import importlib.metadata as importlib_metadata
+except ModuleNotFoundError:
+    import importlib_metadata
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_entry_points(group_name, *, specs=None, strict=False):
+    """
+    Get entry points from a specific group.
+
+    :param str group_name: the name of the entry point group
+    :param list specs: an optional collection of entry point names to retrieve
+    :param bool strict: whether to raise or warn on error
+    :returns: mapping from entry point names to ``EntryPoint`` instances
+    :rtype: dict
+    """
+    if specs is not None:
+        specs = set(specs)
+    entry_points = {}
+    for entry_point in importlib_metadata.entry_points().get(group_name, []):
+        name = entry_point.name
+        if name in entry_points:
+            msg = (f"Found duplicate entry point '{name}': "
+                   'got {entry_point} and {entry_points[name]}')
+            if strict:
+                raise RuntimeError(msg)
+            logger.warning(msg)
+            continue
+        if specs and name not in specs:
```

Comment:
I would recommend putting this check first so that if someone asked for the "c" entrypoint but there were two "cpp" entrypoints then it could still use the "c" extension without being blocked by the problem with the duplicate "cpp" extensions.

Reply:
Sounds like a great idea. See 0128d9d24ccffb1c3d71e402033de8ba1725ab3e.